drivers/flash/soc_flash_nrf.c

```diff
@@ -85,7 +85,11 @@ static inline bool is_aligned_32(u32_t data)
 	return (data & 0x3) ? false : true;
 }
 
+#ifndef CONFIG_SOC_FLASH_NRF_UICR
 static inline bool is_addr_valid(off_t addr, size_t len)
+#else
+static inline bool is_regular_addr_valid(off_t addr, size_t len)
+#endif
```

Comment:
Why complicate things so much?
You could always define this function as `is_regular_addr_valid` and define `is_addr_valid` always in the way you did it for UICR feature enabled, i.e:
```
	return is_regular_addr_valid(addr, len) ||
	       is_uicr_addr_valid(addr, len);
```
Then you would only need to change the implementation of `is_uicr_addr_valid` so that it just returns `false` when the access to UICR is not enabled.
This would also simplify the code in `flash_nrf_erase` as you could go with:
```
if (is_regular_addr_valid(addr, size)) {
	...
	/* check if a proper page erase is requested */
	...

#ifdef CONFIG_SOC_FLASH_NRF_UICR
} else if (addr != (off_t)NRF_UICR || size != sizeof(*NRF_UICR)) {
	return -EINVAL;
}
#else
} else {
	return -EINVAL;
}
#endif
```
The current implementation is not incorrect but it may be a bit hard to understand when some need of further enhancement of this code appears.

Reply:
rewritten